cassandra/src/test/scala/org/apache/zeppelin/cassandra/ParagraphParserTest.scala

```diff
@@ -496,73 +545,180 @@ class ParagraphParserTest extends FlatSpec
     ex.getMessage should be(s"Invalid syntax for DESCRIBE TABLES. It should comply to the pattern: ${DESCRIBE_TABLES_PATTERN.toString}")
   }
 
-  "Parser" should "parse describe keyspace" in {
-    val queries ="Describe KeYsPaCe toto;"
+  "Parser" should "parse describe type" in {
+    val queries ="Describe Type toto;"
 
     val parsed = parser.parseAll(parser.queries, queries)
 
-    parsed.get should be(List(DescribeKeyspaceCmd("toto")))
+    parsed.get should be(List(DescribeTypeCmd(None,"toto")))
   }
 
-  "Parser" should "fail parsing describe keyspace" in {
-    val queries ="Describe KeYsPaCe toto"
+  "Parser" should "parse describe type with keyspace" in {
+    val queries ="Describe Type ks.toto;"
+
+    val parsed = parser.parseAll(parser.queries, queries)
+
+    parsed.get should be(List(DescribeTypeCmd(Some("ks"),"toto")))
+  }
+
+  "Parser" should "fail parsing describe type" in {
+    val queries ="Describe Type toto"
 
     val ex = intercept[InterpreterException] {
       parser.parseAll(parser.queries, queries)
     }
-    ex.getMessage should be(s"Invalid syntax for DESCRIBE KEYSPACE. It should comply to the pattern: ${DESCRIBE_KEYSPACE_PATTERN.toString}")
+    ex.getMessage should be(s"Invalid syntax for DESCRIBE TYPE. It should comply to the patterns: " +
+      s"${DESCRIBE_TYPE_WITH_KEYSPACE_PATTERN.toString} or ${DESCRIBE_TYPE_PATTERN.toString}")
   }
 
-  "Parser" should "parse describe table" in {
-    val queries ="Describe TaBlE toto;"
+  "Parser" should "parse describe types" in {
+    val queries ="Describe types;"
 
     val parsed = parser.parseAll(parser.queries, queries)
 
-    parsed.get should be(List(DescribeTableCmd(None,"toto")))
+    parsed.get(0) shouldBe a [DescribeTypesCmd]
   }
 
-  "Parser" should "parse describe table with keyspace" in {
-    val queries ="Describe TaBlE ks.toto;"
+  "Parser" should "fail parsing describe types" in {
+    val queries ="Describe types"
+
+    val ex = intercept[InterpreterException] {
+      parser.parseAll(parser.queries, queries)
+    }
+    ex.getMessage should be(s"Invalid syntax for DESCRIBE TYPES. It should comply to the pattern: ${DESCRIBE_TYPES_PATTERN.toString}")
+  }
+
+  "Parser" should "parse describe function" in {
+    val queries ="Describe function toto;"
 
     val parsed = parser.parseAll(parser.queries, queries)
 
-    parsed.get should be(List(DescribeTableCmd(Some("ks"),"toto")))
+    parsed.get should be(List(DescribeFunctionCmd(None,"toto")))
```

Comment:
Are these Options?  A more idiomatic way to do this would be
`parsed should be(Some(List(DescribeFunctionCmd(None,"toto"))))`

(and possibly no ( ) after be if not required by compiler)